worker/firewaller/firewaller.go

```diff
@@ -920,11 +930,15 @@ func (fw *Firewaller) machineLifeChanged(tag names.MachineTag) error {
 		return fw.forgetMachine(machined)
 	}
 	if !known && !dead {
-		err = fw.startMachine(tag)
+		watching, err := fw.startMachine(tag)
 		if err != nil {
 			return err
 		}
-		logger.Debugf("started watching %q", tag)
+		if watching {
```

Comment:
If you're only using the bool return for logging, it would be better to log inside `startMachine` and lose the bool (and then you can have different log messages for the different not watching cases).

Reply:
updated.